File: src/tsBusinessLayer/dtos/UserDTO.ts
import {IUser} from "../interfaces/IUser"

export class UserDTO implements IUser{
    public id:number;
    public FirstName:String;
    public LastName:String;
    public Email:String;
    public Address:String;
    public Password: String;
    public Role: { id: number; Name: string; } | undefined;

    constructor(id:number, firstName: String, lastName: String, email: String, address: String, password: String){
        this.id = id;
        this.FirstName = firstName;
        this.LastName = lastName;
        this.Email = email;
        this.Address = address;
        this.Password = password;
    }
}

File: Berix-Library-API/Application/DTOs/Users/UserDTO.cs
using Application.DTOs.Roles;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.DTOs.Users
{
    public class UserDTO
    {
        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }   
        public string Email { get; set; }
        public string Password { get; set; }
        public string Address { get; set; }
        public RoleDTO Role { get; set; }
    }
}
